kt/plugins/godot-intellij-plugin/src/main/kotlin/godot/intellij/plugin/linemarker/SignalConnectionLineMarker.kt

```diff
@@ -0,0 +1,106 @@
+package godot.intellij.plugin.linemarker
+
+import com.intellij.codeInsight.daemon.LineMarkerInfo
+import com.intellij.codeInsight.daemon.LineMarkerProvider
+import com.intellij.openapi.editor.markup.GutterIconRenderer
+import com.intellij.openapi.util.IconLoader
+import com.intellij.psi.PsiElement
+import godot.intellij.plugin.data.cache.SignalConnectionCacheProvider
+import godot.intellij.plugin.extension.isSignal
+import godot.intellij.plugin.ui.dialog.IncomingSignalConnectionsDialog
+import godot.intellij.plugin.ui.dialog.OutgoingSignalConnectionsDialog
+import org.jetbrains.kotlin.nj2k.postProcessing.type
+import org.jetbrains.kotlin.psi.KtFunction
+import org.jetbrains.kotlin.psi.KtProperty
+import org.jetbrains.kotlin.psi.psiUtil.containingClass
+import kotlin.text.Typography.nbsp
+
+class SignalConnectionLineMarker : LineMarkerProvider {
+    override fun getLineMarkerInfo(element: PsiElement): LineMarkerInfo<*>? {
+        if (element is KtProperty && element.type().isSignal()) {
```

Comment:
IIRC for performance reasons, line markers should target leaf nodes: so for this case the leaf node for KtProperty and KtFunction is an identifier (i think). see https://github.com/spekframework/spek/blob/2.x/spek-ide-plugin-intellij-base/src/main/kotlin/org/spekframework/intellij/SpekRunLineMarkerContributor.kt#L24.

Reply:
Indeed wasn't reading the docs carefully enough. Good catch.
Fixed now